### ISSUE TITLE
Investigate feasibility of adding support for Opus audio decoding

### ISSUE THREAD

**Issue by britzl** [defold/defold#9604]
Opus is the successor of Vorbis.

https://opus-codec.org/

![image](https://github.com/user-attachments/assets/486a9453-e4b9-4046-9b5e-e5b4e1398654)

**Comment by aglitchman** [comment]
I would really appreciate it if the team would make audio codecs as extensions before adding Opus. Because I read a long time ago that Opus decoding takes significantly more cpu time, and also Opus loses to Vorbis in file size on short sounds.

A quick search of plus and minus confirms this: https://blog.audiokinetic.com/en/a-guide-for-choosing-the-right-codec/

**Comment by britzl** [comment]
> I would really appreciate it if the team would make audio codecs as extensions before adding Opus.

Yes, this would be a perfect opportunity to have Opus, Ogg and others as plugins/extensions. I'm thinking that Ogg and Wav are the defaults, and that it will be possible to remove and add more decoders. This would make it possible to for instance remove Ogg and only use Wav, or replace Ogg with Opus, or only add Opus as an additional decoder.

**Comment by ShalokShalom** [comment]
I come from the documentation, that left me a bit wondering, that a modern game engine is not supporting the highest class audio codec that is available to us. 

Personally, I consider audio files who are shorter than 200 ms as very rare and the increased CPU usage seems only relevant for very old computers.

The factual CPU usage on my 5800U is 0.1% with mpv. 🤷🏻‍♂️  

One important aspect is also, that opus is capable to enable or disable certain features via compiler switches and runtime switches: https://opus-codec.org/demo/opus-1.5/

My use case is to ship the software on HTML, so size does matter quite a bit. 

I am fully fine with shipping this as an extension, although I think it should be named in the docs, and its also suitable to be used as the default codec, in my opinon. 

Thanks a lot!

**Comment by JCash** [comment]
> I would really appreciate it if the team would make audio codecs as extensions before adding Opus. Because I read a long time ago that Opus decoding takes significantly more cpu time, and also Opus loses to Vorbis in file size on short sounds.

Our decoders are already using the regular extension mechanic, so it's should work well to use an app manifest to add/remove them.

> My use case is to ship the software on HTML, so size does matter quite a bit.
While testing libopus, it also turned out that it was 9% of the html5 build size, which further emphasizes the need to opt-in to opus support.

In the long run, I'd like to have opus as the replacement for ogg vorbis, but I think we need to find a smaller decoder first.
